cocotb/share/lib/gpi_log/gpi_logging.cpp

```diff
@@ -135,132 +197,10 @@ static void gpi_log_native_v(const char *name, enum gpi_log_levels level, const
     fflush(stdout);
 }
 
-static void gpi_log_native(const char *name, enum gpi_log_levels level, const char *pathname, const char *funcname, long lineno, const char *msg, ...)
-{
-    va_list argp;
-    va_start(argp, msg);
-    gpi_log_native_v(name, level, pathname, funcname, lineno, msg, argp);
-    va_end(argp);
-}
-
-/**
- * @name    GPI logging
- * @brief   Write a log message using cocotb SimLog class
- * @ingroup python_c_api
- *
- * GILState before calling: Unknown
- *
- * GILState after calling: Unknown
- *
- * Makes one call to PyGILState_Ensure and one call to PyGILState_Release
- *
- * If the Python logging mechanism is not initialised, dumps to `stderr`.
- *
- */
-static void gpi_log_v(const char *name, enum gpi_log_levels level, const char *pathname, const char *funcname, long lineno, const char *msg, va_list argp)
-{
-    /* We first check that the log level means this will be printed
-     * before going to the expense of formatting the variable arguments
-     */
-    if (!pLogHandler) {
-        gpi_log_native_v(name, level, pathname, funcname, lineno, msg, argp);
-        return;
-    }
-
-    if (level < local_level)
-        return;
```

Comment:
I don't see this in the new code - I think this was a pretty important optimization to avoid too many python/c boundaries

Reply:
I tested just to see what affect this actually has.

#### This branch
total regression wall time: 2m2s
endian swapper example ratio: 6882.48 NS/S

#### Master
rotal regression wall time: 1m55s
endian swapper example ratio: 7716.36 NS/S

So it looks like 10% hit across the board, so it is significant. I'll add this back in. I'm wondering if there is a way to get the Python logger to automatically update the local level if the level is changed in Python.

Reply:
>  I'm wondering if there is a way to get the Python logger to automatically update the local level if the level is changed in Python.

#493 explored doing this, along with a large pile of other stuff. We'd have to go back to providing a non-empty logger subclass, which hooked into `setLevel` calls.

Reply:
The most recent commit added an implementation of this.